js/src/reports/utils.test.js

```diff
@@ -0,0 +1,296 @@
+	} );
+
+	it( 'should sort aggregated intervals in ascending order of string code by `interval`', () => {
+		const intervals1 = toIntervals(
+			[ '2021-01' ],
+			[ '2021-03' ],
+			[ '2021-05' ],
+			[ '2021-07' ]
+		);
+		const intervals2 = toIntervals(
+			[ '2021-02' ],
+			[ '2021-04' ],
+			[ '2021-06' ],
+			[ '2021-08' ]
+		);
+		const expectedIntervals = toIntervals(
+			[ '2021-01' ],
+			[ '2021-02' ],
+			[ '2021-03' ],
+			[ '2021-04' ],
+			[ '2021-05' ],
+			[ '2021-06' ],
+			[ '2021-07' ],
+			[ '2021-08' ]
+		);
+
+		const result = aggregateIntervals( intervals1, intervals2 );
+
+		expectedIntervals.forEach( ( item, i ) => {
+			expect( result[ i ] ).toMatchObject( item );
+		} );
+	} );
+
+	describe( 'aggregate intervals', () => {
+		it( "should merge two given intervals' items by the same `interval`, and the `subtotals` from the same `interval` items should be aggregated by summation of each its metric", () => {
+			const intervals1 = toIntervals(
+				[ '2021-01', 12, 34 ],
+				[ '2021-02', 34, 56 ]
+			);
+			const intervals2 = toIntervals(
+				[ '2021-01', 56, 78 ],
+				[ '2021-02', 78, 90 ]
+			);
+			const expectedIntervals = toIntervals(
+				[ '2021-01', 68, 112 ],
+				[ '2021-02', 112, 146 ]
+			);
+
+			const result = aggregateIntervals( intervals1, intervals2 );
+
+			expect( result ).toHaveLength( expectedIntervals.length );
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+
+		it( "should merge two given intervals' items to a union result by `interval`", () => {
+			const intervals1 = toIntervals( [ '2021-01', 12 ] );
+			const intervals2 = toIntervals( [ '2021-02', 34 ] );
+			const expectedIntervals = toIntervals(
+				[ '2021-01', 12 ],
+				[ '2021-02', 34 ]
+			);
+
+			const result = aggregateIntervals( intervals1, intervals2 );
+
+			expect( result ).toHaveLength( expectedIntervals.length );
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+
+		it( 'When any metric in `paidFields` does not exist in `subtotals`, should fill its value with 0', () => {
+			const intervals1 = toIntervals( [ '2021-01', , 1, , 3 ] );
+			const intervals2 = toIntervals( [ '2021-01', , , 2, , 4 ] );
+			const expectedIntervals = toIntervals( [
+				'2021-01',
+				0,
+				1,
+				2,
+				3,
+				4,
+			] );
+
+			const result = aggregateIntervals( intervals1, intervals2 );
+
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+	} );
+} );
+
+describe( 'sumToPerformance', () => {
+	function toTotals( fields, ...values ) {
+		return values.reduce(
+			( acc, value, i ) => ( {
+				...acc,
+				[ fields[ i ] ]: value,
+			} ),
+			{}
+		);
+	}
+
+	const toFreeTotals = toTotals.bind( null, freeFields );
+	const toPaidTotals = toTotals.bind( null, paidFields );
+
+	it( 'should be able to sum paid `totals` to performance only', () => {
```

Comment:
:nail_care: I find this description quite vague. I don't know what to actually expect.
"be able" is ambiguous to me.

Is it like:
"When only paid totals are given, should map it to PerformanceData object"?

Reply:
Updated by https://github.com/woocommerce/google-listings-and-ads/pull/866/commits/09ac902c6f757a67ff1d266588d6ba9387c6f8a2